airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -147,7 +149,26 @@ def ti_update_state(
             next_kwargs=ti_patch_payload.trigger_kwargs,
             trigger_timeout=timeout,
         )
+    elif isinstance(ti_patch_payload, TIRescheduleStatePayload):
+        task_instance = session.scalar(select(TI).where(TI.id == ti_id_str))
+        actual_start_date = timezone.utcnow()
+        # update the task_reschedule table first
+        session.add(
+            TaskReschedule(
```

Comment:
Are we creating a new TaskReschedule here or updating as per comment on line 155?

Reply:
Should be better now @ephraimbuddy
Handled in 53c80679d878ad2f6ad9f70fb4f2304176591b1e